feat: Add retrieving Transaction (#1775)
getsentry/sentry-docs: src/collections/_documentation/performance/distributed-tracing.md

while True:
  item = get_from_queue()

>   with sentry_sdk.startTransaction(op="task", name=item.get_transaction()):

Reviewer: ```suggestion
  with sentry_sdk.start_transaction(op="task", name=item.get_transaction()):
```

      span.set_data("http.foobarsessionid", get_foobar_sessionid())
```

#### Retrieving a Transaction

> TODO

Reviewer: This is going to be along the lines of:

```py
from sentry_sdk import Hub

transaction = Hub.current.scope.transaction

# ... start child span or new transaction depending whether `transaction is None`
```

}
```

> #### Retrieving a Transaction

Reviewer: The same heading appearing multiple times in the same page (3x in this page) will all generate the same anchor link.

Do we have some helper to customize the anchor name? We could use that to make those 3 headings use different anchors.

> In cases where you want to attach Spans to an already ongoing Transaction you can use `Sentry.getCurrentHub().getScope().getTransaction()`. This function will return a `Transaction` in case there is a running Transaction otherwise it returns `undefined`. If you are using our Tracing integration by default we attach the Transaction to the Scope. So you could do something like this:

Reviewer 2: ```suggestion
In cases where you want to attach a span to already ongoing transaction, you can use `Sentry.getCurrentHub().getScope().getTransaction()` call. This function will return a `Transaction` in case there is a running transaction or `undefined` otherwise. If you are using our Tracing integration, we attach the transaction to the scope by default, so you could do something like this:
```

```javascript
function myJsFunction() {

>   const transaction = Sentry.getCurrentHub().getScope().getTransaction();  

Reviewer: Wouldn't this be a good case to use `withScope`?

```js
func f() {
  Sentry.withScope((scope) => {
    const transaction = scope.getTransaction();
    let span;
    if (transaction) {
      span = transaction.startChild({
        op: "encode",
        description: "parseAvatarImages"
      });
    } else {
      span = hub.startTransaction({
        name: "f",
        op: "encode",
        description: "parseAvatarImages"
      });
    }
    try {
      // Do work
    } finally {
      span.finish();
    }
  });
}

// did not test, may contain typos; typed in GitHub directly.
```

Reviewer 2: `withScope` creates a new scope, but we could use `configureScope` here just fine imo.

```js
Sentry.configureScope(scope => {
  const transaction = scope.getTransaction();
  if (transaction) {
    let span = transaction.startChild({
      op: "encode",
      description: "parseAvatarImages"
    });
    // Do something
    span.finish();
  }
});

});
```

#### Retrieving a Transaction

> In cases where you want to attach Spans to an already ongoing Transaction you can use `Sentry.getCurrentHub().getScope().getTransaction()`. This function will return a `Transaction` in case there is a running Transaction otherwise it returns `undefined`. If you are using our Express integration by default we attach the Transaction to the Scope. So you could do something like this:

Reviewer 2: ```suggestion
In cases where you want to attach a span to already ongoing transaction, you can use `Sentry.getCurrentHub().getScope().getTransaction()` call. This function will return a `Transaction` in case there is a running transaction or `undefined` otherwise. If you are using our Tracing integration, we attach the transaction to the scope by default, so you could do something like this:
```

```javascript
app.get("/success", function successHandler(req, res) {
  const transaction = Sentry.getCurrentHub().getScope().getTransaction();
  if (transaction) {
    let span = transaction.startChild({
      op: "encode",
      description: "parseAvatarImages"
    });
    // Do something

>     span.finish();

Reviewer: In Python we have context managers. The way they work guarantees that even if there is an exception in the middle of a span, the span will be finished.

Without `withTransaction` and `withSpan` wrappers, it might be a good idea to document using `try...finally` and calling `finish()` in the `finally` block.
